Formula/m/msieve.rb

```diff
@@ -0,0 +1,22 @@
+class Msieve < Formula
+  desc "C library for factoring large integers"
+  homepage "https://sourceforge.net/projects/msieve/"
+  url "https://downloads.sourceforge.net/project/msieve/msieve/Msieve%20v1.53/msieve153_src.tar.gz"
+  sha256 "c5fcbaaff266a43aa8bca55239d5b087d3e3f138d1a95d75b776c04ce4d93bb4"
+  license :public_domain
+
+  livecheck do
+    url :stable
+    regex(%r{url=.*?/Msieve%20v?(\d+(?:\.\d+)+)/}i)
+  end
+
+  def install
+    inreplace "common/util.c", "#include <util.h>", "#include <util.h>\n#include <sys/time.h>"
+    system "make", "all"
+    bin.install "msieve"
+  end
+
+  test do
+    system "#{bin}/msieve", "-h"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
checkout the other c library for some test inspirations.

Reply:
It seemed complicated to write a test for the library, so I rewrote it as a simple test for the CLI, is that OK?